akka-docs/src/main/paradox/typed/interaction-patterns.md

```diff
@@ -89,65 +128,87 @@ The function is running in the receiving actor and can safely access state of it
 
 **Useful when:**
 
  * Single response queries
- * When an actor needs to know that the message was processed before continuing 
+ * An actor needs to know that the message was processed before continuing 
  * To allow an actor to resend if a timely response is not produced
- * To keep track of outstanding requests and not overwhelm a recipient with messages (simple backpressure)
- * When some context should be attached to the interaction but the protocol does not support that (request id, what query the response was for)
+ * To keep track of outstanding requests and not overwhelm a recipient with messages ("backpressure")
+ * Context should be attached to the interaction but the protocol does not support that (request id, what query the response was for)
  
 **Problems:**
 
- * There can only be a single response to one `ask`
+ * There can only be a single response to one `ask` (see @ref:[per session child Actor](#per-session-child-actor))
  * When `ask` times out, the receiving actor does not know and may still process it to completion, or even start processing it after the fact
- * Finding a good value for the timeout, especially when `ask` is triggers chained `ask`s in the receiving actor. You want a short timeout to be responsive and answer back to the requestor, but at the same time you do not want to have many false positives 
+ * Finding a good value for the timeout, especially when `ask` is triggers chained `ask`s in the receiving actor. You want a short timeout to be responsive and answer back to the requester, but at the same time you do not want to have many false positives 
+
 
+## Request-Response with ask from outside the ActorSystem
 
-## 1:1 Request-Response with ask from outside the ActorSystem
+Some times you need to interact with actors from outside of the actor system, this can be done with fire-and-forget as described above or through another version of `ask` that returns a @scala[`Future[Response]`]@java[`CompletionStage<Response>`] that is either completed with a succesful response or failed with a `TimeoutException` if there was no response within the specified timeout. 
+ 
+To do this we use @scala[`ActorRef.ask` (or the symbolic `ActorRef.?`) implicitly provided by `akka.actor.typed.scaladsl.AskPattern`]@java[`akka.actor.typed.javadsl.AskPattern.ask`] to send a message to an actor and get a @scala[`Future[Response]`]@java[`CompletionState[Response]`] back.
 
-In an interaction where there is a 1:1 mapping between a request and a response we can use @scala[`ActorRef.?` implicitly provided by `akka.actor.typed.scaladsl.AskPattern`]@java[`akka.actor.typed.javadsl.AskPattern.ask`] to send a message to an actor and get a @scala[`Future[Response]`]@java[`CompletionState[Response]`] back.
+Scala
+:  @@snip [InteractionPatternsSpec.scala]($akka$/akka-actor-typed-tests/src/test/scala/docs/akka/typed/InteractionPatternsSpec.scala) { #standalone-ask }
```

Comment:
It would be good to show the curried version for better type inference, so add another field to GiveMeCookies

Reply:
I'm not sure I follow, add another field and have respondTo last and use `GiveMeCookies.apply.curried(param)`? Feels a lot messier as a sample to me.

Reply:
E.g.
```
case class Request(correlationId: Int, data: String)(sender: ActorRef[Response])
```
And then you don't need any type annotations in a statement like:

```
sideEffectProcessor ? Request(correlationId, data)
```

Otherwise you need to type what this is being assigned to or the _ passed to the Request constructor

Reply:
I solved this by using the explicit lambda instead. Fancy-pants-trix can be left as an exercise to the reader.